### PR TITLE
Parametrize backup/restore tests to run on capsule

### PR DESCRIPTION
**Pull request by vsedmik**
Adding parametrization of the f-m backup/restore tests to be run against the Capsule as well.

### CODE REVIEW

**Comment by vsedmik** [comment]
trigger: test-robottelo
pytest: tests/foreman/maintain/test_backup_restore.py

**Comment by vsedmik** [comment]
It seems the PRT struggles with broker checkout. Local run results are:
```
$ pytest tests/foreman/maintain/test_backup_restore.py
====================================== test session starts ========================================
platform linux -- Python 3.10.8, pytest-7.2.0, pluggy-1.0.0

collected 43 items / 1 deselected / 42 selected                                                                                                                                                                                        

tests/foreman/maintain/test_backup_restore.py ..........................................    [100%]

=================== 42 passed, 1 deselected, 97 warnings in 13276.09s (3:41:16) ===================
```

**Comment by vsedmik** [comment]
Rebased.

File: tests/foreman/maintain/test_backup_restore.py [hunk 1]
```diff
@@ -421,17 +488,24 @@ def test_positive_backup_restore(
...
-        expected_files = BASIC_FILES | REMOTE_SAT_FILES
+    if type(sat_maintain) is Satellite:
+        if sat_maintain.is_remote_db():
+            expected_files = BASIC_FILES | REMOTE_SAT_FILES
+        else:
+            expected_files = (
+                BASIC_FILES | OFFLINE_FILES
+                if backup_type == 'offline'
+                else BASIC_FILES | ONLINE_SAT_FILES
+            )
```
Comment (Griffin-Sullivan) [inline tests/foreman/maintain/test_backup_restore.py:503]: Could we make this a helper function? Looks like this logic is repeated a couple times.

**Review by Griffin-Sullivan — APPROVED** [review]
Looks good! One non-blocking comment.

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/maintain/test_backup_restore.py

File: tests/foreman/maintain/test_backup_restore.py (hunk 1 above)
Comment (vsedmik) [reply tests/foreman/maintain/test_backup_restore.py]: Sure thing :+1:

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/maintain/test_backup_restore.py

File: tests/foreman/maintain/test_backup_restore.py [hunk 2]
```diff
@@ -254,23 +274,29 @@ def test_positive_backup_offline_logical(sat_maintain, setup_backup_tests, modul
...
 
     # Check for expected files
-    backup_dir = re.findall(fr'{subdir}\/satellite-backup-.*-[0-5][0-9]', result.stdout)[0]
+    backup_dir = re.findall(fr'{subdir}\/{instance}-backup-.*-[0-5][0-9]', result.stdout)[0]
     files = sat_maintain.execute(f'ls -a {backup_dir}').stdout.split('\n')
     files = [i for i in files if not re.compile(r'^\.*$').search(i)]
 
-    if sat_maintain.is_remote_db():
-        expected_files = BASIC_FILES | REMOTE_SAT_FILES
+    if type(sat_maintain) is Satellite:
```
Comment (pondrejk) [inline tests/foreman/maintain/test_backup_restore.py:281]: is there a reason not to use `get_expected_files` also here

**Review by pondrejk — APPROVED** [review]
ack pending one comment and a rebase

File: tests/foreman/maintain/test_backup_restore.py (hunk 2 above)
Comment (vsedmik) [reply tests/foreman/maintain/test_backup_restore.py:281]: yes, in this particular case we expect files for both backup types created

**Comment by vsedmik** [comment]
Rebased.

**Pull request merged**